common/lib/internal_pooled_connection_provider.ts

```diff
@@ -39,14 +38,15 @@ import { AwsPoolConfig } from "./aws_pool_config";
 import { LeastConnectionsHostSelector } from "./least_connections_host_selector";
 import { PoolClientWrapper } from "./pool_client_wrapper";
 import { logger } from "../logutils";
+import { SlidingExpirationCacheWithCleanupTask } from "./utils/sliding_expiration_cache_with_cleanup_task";
 
 export class InternalPooledConnectionProvider implements PooledConnectionProvider, CanReleaseResources {
   static readonly CACHE_CLEANUP_NANOS: bigint = BigInt(10 * 60_000_000_000); // 10 minutes
   static readonly POOL_EXPIRATION_NANOS: bigint = BigInt(30 * 60_000_000_000); // 30 minutes
-  protected static databasePools: SlidingExpirationCache<string, any> = new SlidingExpirationCache(
+  protected static databasePools: SlidingExpirationCacheWithCleanupTask<string, AwsPoolClient> = new SlidingExpirationCacheWithCleanupTask(
     InternalPooledConnectionProvider.CACHE_CLEANUP_NANOS,
-    (pool: any) => pool.getActiveCount() === 0,
-    (pool: any) => pool.end()
+    (pool: AwsPoolClient) => pool.getActiveCount() === 0,
+    async (pool: AwsPoolClient) => await pool.end()
```

Comment:
Is pool.releaseResources() necessary here? It used to be in the releaseResources method